flow/templates/base_script.sh

```diff
@@ -17,7 +17,7 @@ cd {{ project.config.project_dir }}
 {% set cmd_suffix = cmd_suffix|default('') ~ (' &' if parallel else '') %}
 {% for operation in operations %}
 {% if operation.directives.nranks and not mpi_prefix %}
-{% set mpi_prefix = "%s -n %d "|format(mpiexec|default("mpiexec"), operation.directives.nranks) %}
+{% set mpi_prefix = operation|generate_mpi_prefix %}
```

Comment:
I think we should not specify a separate variable here, but instead line 27 should read:

`{{ mpi_prefix|default(operation|generate_mpi_prefix }}`

In this way we still respect any `mpi_prefix` constants that the user provided manually, but otherwise get the prefix from the environment.